packages/salesforcedx-vscode-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -226,71 +222,34 @@ export class CheckpointService implements TreeDataProvider<BaseNode> {
       theNode.setActionCommandResultId(result.id);
       return;
     }
-    // Failure could mean that there is already an entry with the same the criteria.
-    // If that's the case then pase the Id out, delete it and recreate it.
-    // Fun fact: the result is an array of 1 item
-    // "[{"message":"duplicate value found: ScopeId duplicates value on record with id: 1doxx00000000Pn","errorCode":"DUPLICATE_VALUE","fields":[]}]"
-    // Fun fact #2: The result message can also be an outright failure string. For example if we can't connnect up to the AppServer then the result
-    // is a plain string. It'll cause the parsing to fail.
+    // Fun fact: the result is an array of 1 item OR the result message can be just a string. In the
+    // case where the json string cannot be parsed into an ApexExecutionOverlayFailureResult then it'll
+    // be treated as a string and reported to the user.
     if (errorString) {
       try {
         const result = JSON.parse(
           errorString
         ) as ApexExecutionOverlayFailureResult[];
-        // If the node is a dupe, then delete it and recreate it. If the failure reason
-        // was something else that we don't handle then just set the command result.
-        if (result[0].errorCode === DUPLICATE_VALUE && recreateIfDupe) {
-          // parse the record ID from the result[0].message, call remove and then call add again
-          const duplicateId = result[0].message
-            .substr(result[0].message.lastIndexOf(':') + 1)
-            .trim();
-          // set the id to the duplicate id and call the delete to nuke the record,
-          // then call the delete again with the recreation flag unset
-          theNode.setActionCommandResultId(duplicateId);
-          await this.executeRemoveApexExecutionOverlayActionCommand(theNode);
-          // clear out the action command resultId
-          theNode.setActionCommandResultId(undefined);
-          await this.executeCreateApexExecutionOverlayActionCommand(
-            theNode,
-            false /* recreateIfDupe */
+        if (result[0].errorCode === FIELD_INTEGRITY_EXCEPTION) {
+          vscode.window.showErrorMessage(
+            nls.localize('local_source_is_out_of_sync_with_the_server')
           );
-        // This is the error code we'll see when the local source and the server are out of sync
-        // This means that the local type does not exist on the server. Give the error a more useful
-        // message as to why they can't set the checkpoint.
-        } else if (result[0].errorCode === FIELD_INTEGRITY_EXCEPTION) {
-          theNode.setActionCommandResultFailure(nls.localize('local_source_is_out_of_sync_with_the_server'));
         } else {
-          theNode.setActionCommandResultFailure(result[0].message);
+          vscode.window.showErrorMessage(
```

Comment:
You can use template strings: `${result[0].message}. URI=${theNode.getCheckpointUri()}, Line=${theNode.getCheckpointLineNumber()}`

Reply:
I'll make the change but here's one reason I don't like template strings. Say you're changing a getCheckpointUri to a different name, Find All References will not find the references that are in the template strings.

Reply:
It works for me.

![image](https://user-images.githubusercontent.com/7286437/39770966-744b1d28-52a5-11e8-9609-1afb16b0850b.png)

What do you see?

Reply:
The wrong single quotes on mine. You are correct.